### ISSUE TITLE
Option to show sample output & other info while training

### ISSUE THREAD

**Issue by antirez** [Om-Alve/smolGPT#19]
Thanks for the project! In the past I played with simple language models to run in my computer, and since training is slow and tiring, I found that it becomes much more interesting if from time to time I output informations such as loss, an example of output generation, update graphs with the training loss curve, and so forth. It becomes a lot more interesting and makes the casual user more willing to wait for hours.

**Comment by Om-Alve** [comment]
Thanks a lot, This is a great idea! Will implement this as soon as possible!

**Comment by Om-Alve** [comment]
How does this look? 
![Image](https://github.com/user-attachments/assets/b5ddd5c4-5d26-4d7e-ba3c-83b178496c4b)w

**Comment by Om-Alve** [comment]
Right now, I'm logging the losses to `tensorboard` in the `logs` folder, you can check the training loss and validation loss curves there, Would love to know if i could add anything more!

**Comment by Om-Alve** [comment]
> I'm logging the losses to tensorboard in the logs folder

I should probably add this in the readme as well

**Comment by antirez** [comment]
> How does this look? ![Image](https://github.com/user-attachments/assets/b5ddd5c4-5d26-4d7e-ba3c-83b178496c4b)w

That's awesome! Thank you :)

**Comment by Om-Alve** [comment]
Done, Have pushed it! I'm honored to have you review my project. Thank you!

**Comment by antirez** [comment]
Thank you @Om-Alve! Really appreciated :) Now that AI is so important, people may feel a total detatch between the large language models we use every day and the technology itself. Every time it's possible I encourage people to train a small language model, understand its implementation and so forth, as this closes the gap, so I believe projects like that, of small LLMs, are super important. Especially I like that because it uses advanced tokenization and positional encoding and so forth, so it's a lot more similar to the "true thing" compared to GPT-2 or simpler language models. So thank you :)

**Comment by Om-Alve** [comment]
This made my day 😄
